python/lsst/analysis/tools/atools/cpMetrics.py

```diff
@@ -0,0 +1,223 @@
+
+    def addFpPlot(self, vectorKey, statistic, label):
+        self.process.buildActions.x = LoadVector()
+        self.process.buildActions.x.vectorKey = "detector"
+        self.process.buildActions.y = LoadVector()
+        self.process.buildActions.y.vectorKey = "amplifier"
+        self.process.buildActions.detector = LoadVector()
+        self.process.buildActions.detector.vectorKey = "detector"
+        self.process.buildActions.amplifier = LoadVector()
+        self.process.buildActions.amplifier.vectorKey = "amplifier"
+        self.process.buildActions.z = LoadVector()
+        self.process.buildActions.z.vectorKey = vectorKey
+        self.process.buildActions.statMask = VectorSelector()
+        self.process.buildActions.statMask.vectorKey = "statMask"
+
+        self.produce.plot = FocalPlaneGeometryPlot()
+        self.produce.plot.statistic = statistic
+        self.produce.plot.xAxisLabel = "x (focal plane)"
+        self.produce.plot.yAxisLabel = "y (focal plane)"
+        self.produce.plot.zAxisLabel = label
+
+
+class BiasScalarMetrics(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addPair("MEAN", "mean", "bias_residual_mean")
+        self.addPair("NOISE", "noise", "bias_residual_noise")
+        self.addPair("CR_NOISE", "crnoise", "bias_residual_crnoise")
+        self.addPair("READ_NOISE", "rn", "bias_read_noise")
+
+        self.produce.metric.units = self.units
+        self.produce.metric.newNames = self.newNames
+
+
+class BiasResidualMeanMedianFP(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addFpPlot("MEAN", "median", "median per-amplifier residual mean (ADU)")
+
+
+class BiasResidualMeanStdevFP(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addFpPlot("MEAN", "std", "stdev per-amplifier residual mean (ADU)")
+
+
+class BiasReadNoiseMedianFP(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addFpPlot("READ_NOISE", "median", "median read noise (ADU)")
+
+
+class BiasReadNoiseStdevFP(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addFpPlot("READ_NOISE", "std", "median read noise (ADU)")
+
+
+class DarkScalarMetrics(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addPair("MEAN", "mean", "dark_residual_mean")
+        self.addPair("NOISE", "noise", "dark_residual_noise")
+        self.addPair("CR_NOISE", "crnoise", "dark_residual_crnoise")
+        self.addPair("READ_NOISE", "rn", "dark_read_noise")
+
+        self.produce.metric.units = self.units
+        self.produce.metric.newNames = self.newNames
+
+
+class DarkResidualMeanMedianFP(CalibrationMetric):
+    def setDefaults(self):
+        super().setDefaults()
+        self.addFpPlot("MEAN", "median", "median per-amplifier residual mean (ADU)")
```

Comment:
Are there no calib-type-specific labels for the plot? I guess the plot title comes from the dataset.